### PR TITLE
(RHEL-32494) ci: deploy systemd man to GitHub Pages

### PR DESCRIPTION
**Pull request by jamacku**
rhel-only

Related: RHEL-1087

### CODE REVIEW

**Comment by dtardon** [comment]
@jamacku I guess I'm missing the context here. I understand what the action does, but not why do we want it 🤔

**Comment by jamacku** [comment]
@dtardon, it was requested by @lnykryn. So we would have an easy way to access the `systemd.net-naming-scheme` man page.

I have already done the same for rhel9, so I thought, why not add it also to rhel8. 
https://redhat-plumbers.github.io/systemd-rhel9/

File: .github/workflows/deploy-man-pages.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          RELEASE="$(lsb_release -cs)"
+          sudo bash -c "echo 'deb-src http://archive.ubuntu.com/ubuntu/ $RELEASE main restricted universe multiverse' >>/etc/apt/sources.list"
+          sudo add-apt-repository -y ppa:upstream-systemd-ci/systemd-ci
+          sudo apt-get -y update
+          sudo apt-get -y build-dep systemd
+          sudo apt-get install -y \
```
Comment (mrc0mmand) [inline .github/workflows/deploy-man-pages.yml]: This installs a lot of really unnecessary (test-only) deps. The build-dep above should be more than enough for building man pages.

File: .github/workflows/deploy-man-pages.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          RELEASE="$(lsb_release -cs)"
+          sudo bash -c "echo 'deb-src http://archive.ubuntu.com/ubuntu/ $RELEASE main restricted universe multiverse' >>/etc/apt/sources.list"
+          sudo add-apt-repository -y ppa:upstream-systemd-ci/systemd-ci
```
Comment (mrc0mmand) [inline .github/workflows/deploy-man-pages.yml]: Not necessary.

**Review by lnykryn — APPROVED** [review]

File: .github/workflows/deploy-man-pages.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      pages: write
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          RELEASE="$(lsb_release -cs)"
+          sudo bash -c "echo 'deb-src http://archive.ubuntu.com/ubuntu/ $RELEASE main restricted universe multiverse' >>/etc/apt/sources.list"
```
Comment (mrc0mmand) [inline .github/workflows/deploy-man-pages.yml]:
```suggestion
          sudo add-apt-repository -y --no-update --enable-source
```

**Review by mrc0mmand — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-man-pages.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            python3-setuptools    \
+            quota                 \
+            strace                \
+            unifont               \
+            util-linux            \
+            zstd                  \
+
+      - name: Build HTML man pages
+        run: |
+          meson build
```
Comment (mrc0mmand) [inline .github/workflows/deploy-man-pages.yml]: Maybe change this to `meson setup build`, newer meson versions complain otherwise.

**Comment by jamacku** [comment]
@mrc0mmand Thanks for review. I needed to make it work fast for Lukaš to share it. I'll try to minimize the list of dependencies.

**Comment by dtardon** [comment]
> @dtardon, it was requested by @lnykryn. So we would have an easy way to access the `systemd.net-naming-scheme` man page.

Such way does already exist: `./build/man/man` and `./build/man/html`. These commands build and display a specified man page, e.g., `./build/man/man systemd.net-naming-scheme`. They have the additional advantage that one can see the effect of local changes immediately.

**Comment by jamacku** [comment]
true, but it was for sharing with others, as a reference for technical writers.

**Comment by jamacku** [comment]
Oops, for some reason, the origin was pointing to upstream :disappointed:

File: .github/workflows/deploy-man-pages.yml (hunk 1 above)
Comment (jamacku) [reply .github/workflows/deploy-man-pages.yml]: Thank you for review @mrc0mmand

**Pull request merged**